### ISSUE TITLE
Troubleshoot SAML problems logging into load-balanced stage

### ISSUE THREAD

**Issue by sfisher** [CDL-Dryad/dryad-product-roadmap#411]
We get the following error trying to log in to stage with shibboleth/SAML.

```
opensaml::BindingException
The system encountered an error at Fri Aug 2 14:53:07 2019

To report this problem, please contact the site administrator at uc3@ucop.edu.

Please include the following message in any email:

opensaml::BindingException at (https://dryadx2-stg.cdlib.org/Shibboleth.sso/SAML2/POST)

SAML message delivered with POST to incorrect server URL.
```

The server doesn't match the front door, ALB server `dryad-stg.cdlib.org` .  Maybe we eitehr need to add each worker to SAML as registered callbacks or else maybe the Shibboleth daemon running on the server had a problem.

I traced the login path with the chrome debugger.  The URL we give to shibboleth looks like `https://dryad-stg.cdlib.org/Shibboleth.sso/Login?target=https%3A%2F%2Fdryad-stg.cdlib.org%2Fstash%2Fauth%2Fshibboleth%2Fcallback%3Ftenant_id%3Ducop&entityID=urn%3Amace%3Aincommon%3Aucop.edu`.  I believe this target for the post-login URL is correct for the application.

The final URL (after the login forms, etc) is `https://dryad-stg.cdlib.org/Shibboleth.sso/SAML2/POST`.  IDK why this doesn't work.

@briri Did you see anything like this when setting up load-balanced login with the dmptool and ALB?  Could this be a shibboleth daemon problem or something else?  Or do we also need to add each worker to shibboleth as a registered callback url?

**Comment by sfisher** [comment]
I looked at the dmptool shibboleth config in `shibboleth2.xml` and I see some stuff in there that we don't have.  Specifically I see a `RequestMapper` section like

```
    <RequestMapper type="Native">
        <RequestMap applicationId="default">
            <Host name="dmptool-stg.cdlib.org" applicationId="dmptool-stg" />
            <Host name="dmp-stage.cdlib.org" applicationId="dmp-stage" />
        </RequestMap>
    </RequestMapper>
```

There are also some other bits that might be different.

Brian said they didn't need to register each instance of dmptool for the UI workers in shibboleth, so there must be something that might get this working.  I also came across these pages which I think mentions how to get an application that spans multiple servers working.  Not sure if it's helpful, but maybe.

https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPApplicationModel

https://wiki.shibboleth.net/confluence/display/SHIB2/NativeSPApplicationOverride
